src/main/java/br/com/sbk/sbking/gui/JElements/CardButton.java

```diff
@@ -32,15 +38,31 @@ public CardButton(Card card, DeckCardImageInformation deckCardImageInformation)
 		this.setMouseListener(this);
 	}
 
+	private void setCardAsSelected() {
+		if (this.faceUp && isInHand) {
```

Comment:
This could be refactored into a validation method like canBePlayed().
Also, the code is very similar between the if and else. I think it may be refactored to something like:
```
if( this.canBePlayed() ){
   this.moveVertically(offset);
}else{
   this.moveVertically(-offset);
}
```

Reply:
I think canBePlayed is a bad name for a method under class Card. The name wrongly suggests that this evaluates the card against some defined rule set (like in "can I play this card on this trick?") which is not what this boolean expression is evaluating.

I also think (faceUp && isInHand) is small enough to not need refactoring. I did however moved the contents of the offset translation to a translate method!